parsec/event_bus.py

```diff
@@ -15,40 +33,54 @@ class MetaEvent(Enum):
     EVENT_DISCONNECTED = "event.disconnected"
 
 
+CustomEvent = TypeVar("CustomEvent", bound=Enum)
+AllEvent = Union[MetaEvent, CustomEvent]
+
+
+class EventCallback(Protocol):
+    def __call__(self, event: Enum, **kwargs: Any) -> None:
+        ...
+
+
+class EventFilterCallback(Protocol):
+    def __call__(self, event: Enum, **kwargs: Any) -> bool:
+        ...
+
+
 class EventWaiter:
-    def __init__(self, filter):
+    def __init__(self, filter: Optional[EventFilterCallback]):
         self._filter = filter
         self._event_occured = trio.Event()
-        self._event_result = None
+        self._event_result: Optional[Tuple[Enum, Dict[str, Any]]] = None
 
-    def _cb(self, event: Enum, **kwargs):
+    def _cb(self, event: Enum, **kwargs: Any) -> None:
         if self._event_occured.is_set():
             return
         if self._filter and not self._filter(event, **kwargs):
             return
         self._event_result = (event, kwargs)
         self._event_occured.set()
 
-    async def wait(self):
+    async def wait(self) -> Tuple[Enum, Dict[str, Any]]:
         await self._event_occured.wait()
-        return self._event_result
+        return self._event_result  # type: ignore
```

Comment:
Use `assert self._event_result is not None` instead to tell mypy that the `None` type should not be considered from this point on.

Reply:
👍

Reply:
cf7d9408ba8492afd17f56a7de61931e5780c0c1